Add SBF documentation (#1830)
PX4/PX4-user_guide: en/gps_compass/septentrio_asterx-rib.md

# AsteRx OEM with Robotics Interface Board

Septentrio is the leading worldwide supplier of OEM GPS / GNSS receivers.
Septentrio OEM receivers deliver accurate and reliable positions to demanding industrial applications in a small, light-weight form factor. 

> We also offer several dual antenna receivers providing heading & pitch or heading & roll angles ideal for autonomous navigation or machine control.

Reviewer: ```suggestion
We also offer several dual antenna receivers which allows for GPS information to be fused into the heading (other attitude information can also be determined, but is not fused by PX4).

```

# AsteRx OEM with Robotics Interface Board

> Septentrio is the leading worldwide supplier of OEM GPS / GNSS receivers. Our OEM receivers deliver accurate and

Reviewer 2: This should IMO not refer to as 'our'.

Author: ```suggestion
Septentrio is the leading worldwide supplier of OEM GPS / GNSS receivers. Septentrio OEM receivers deliver accurate and

```

Author: Flight log:
https://logs.px4.io/plot_app?log=ed164985-b9eb-4f50-80ec-55de2e83a2c5 

Do i need to test something specific for the flight log?

Reviewer: @SeppeG I suspect @bkueng Wants to be able to see that the position uorb topics are being populated correctly and at the right rate. Further that the vehicle holds behaves as expected in a mode that relies on position.

Upshot, I would demonstrate a short mission and also a little manual flight in position mode.

Reviewer: PS, yes, like that. I will commit your suggestion.

Reviewer 2: > @SeppeG I suspect @bkueng Wants to be able to see that the position uorb topics are being populated correctly and at the right rate. Further that the vehicle holds behaves as expected in a mode that relies on position.

Upshot, I would demonstrate a short mission and also a little manual flight in position mode.

Exactly, just that we're sure it's working as expected.

19.7in or more)

For additional configuration of the dual antenna setup, please refer to
our [Knowledge Base](https://support.septentrio.com/l/858493/2022-04-19/xgrsh) or the [hardware manual](https://web.septentrio.com/l/858493/2022-04-19/xgrsl)
:::

### Web app

mosaic-H GPS/GNSS receiver module with heading comes with fully documented interfaces, commands and data messages. The
included GNSS receiver control and analysis software [RxTools](https://web.septentrio.com/l/858493/2022-04-19/xgrss)
allows receiver configuration, monitoring as well as data logging and analysis.

The receiver includes an intuitive web user interface for easy operation and monitoring allowing you to control the
receiver from any mobile device or computer. The web interface also uses easy-to-read quality indicators ideal to
monitor the receiver operation during the job at hand.

![Illustrative image for Septentrio mosaic-H GNSS Receiver Module Web User Interface (WebUI)]( ../../assets/hardware/gps/septentrio_sbf/Septentrio-mosaic-X5-H-T-CLAS-GNSS-Module-Receiver-WebUI.png "Septentrio web user interface")

## PX4 setup

> ![QGroundControl parameter settings](../../assets/hardware/gps/septentrio_sbf/QGC_param.png "QGroundControl parameter settings")

Reviewer: This still looks like it is being set up to point to TELEM1

# AsteRx OEM with Robotics Interface Board

Septentrio is the leading worldwide supplier of OEM GPS / GNSS receivers. Our OEM receivers deliver accurate and
reliable positions to demanding industrial applications in a small, light-weight form factor. We also offer several dual
antenna receivers providing heading & pitch or heading & roll angles ideal for autonomous navigation or machine control.
Inertial sensor integration of the AsteRx-i family offers a full attitude solution (heading, pitch and roll)
synchronized with accurate positioning.

The Robotics Interface Board coupled with Septentrio GNSS receiver boards provides common interfaces like USB, ethernet,
on board logging and other functionalities designed for rapid prototyping, product evaluation or efficient integration.

> Size: 71.53 x 47.5 x 18.15 mm 

Reviewer: Put both of these under Key Features heading as you did in the other doc?

> Weight = 50g

Reviewer: Move these under "Key Features" heading?

> <div style="text-align: center;">
> 
> ![](../../assets/hardware/gps/septentrio_sbf/AsteRX-m3+RIB-board.png "Septentrio Robotics Interface Board")
> 
> </div>

Reviewer: ```suggestion
![Septentrio Robotics Interface Board](../../assets/hardware/gps/septentrio_sbf/AsteRX-m3+RIB-board.png)
```

> <div style="text-align: center;">
> 
> ![](../../assets/hardware/gps/septentrio_sbf/AsteRX-m3+RIB-board.png "Septentrio Robotics Interface Board")
> 
> </div>

Reviewer: ```suggestion
![Septentrio Robotics Interface Board](../../assets/hardware/gps/septentrio_sbf/AsteRX-m3+RIB-board.png)
```

> ## Table Of Contents
> 1. [Key Features](../../en/gps_compass/septentrio_asterx-rib.md#key-features)
> 2. [Purchase](../../en/gps_compass/septentrio_asterx-rib.md#purchase)
> 3. [Interfaces](../../en/gps_compass/septentrio_asterx-rib.md#interfaces)
>    1. [USB](../../en/gps_compass/septentrio_asterx-rib.md#usb)
>    2. [44-pin Header](../../en/gps_compass/septentrio_asterx-rib.md#44-pin-header)
>    3. [LED's](../../en/gps_compass/septentrio_asterx-rib.md#leds)
>    4. [LOG Button Header](../../en/gps_compass/septentrio_asterx-rib.md#log-button-header)
>    5. [PPS/Event Header](../../en/gps_compass/septentrio_asterx-rib.md#ppsevent-header)
>    6. [Power Supply Options](../../en/gps_compass/septentrio_asterx-rib.md#power-supply-options)
> 4. [Hardware Setup](../../en/gps_compass/septentrio_asterx-rib.md#hardware-setup)
>    1. [Dual Antenna](../../en/gps_compass/septentrio_asterx-rib.md#hardware-setup)
>    2. [Web App](../../en/gps_compass/septentrio_asterx-rib.md#web-app)
> 5. [PX4 Setup](../../en/gps_compass/septentrio_asterx-rib.md#px4-setup)
>    1. [Single Antenna](../../en/gps_compass/septentrio_asterx-rib.md#single-antenna)
>    2. [Dual Antenna](../../en/gps_compass/septentrio_asterx-rib.md#px4-setup)

Reviewer: You don't need this. Auto generated by docs toolchain.

```suggestion
```

- On-board logging
- Micro USB connector

## Purchase

All AsteRx receivers and Robotic Interface Boards can be purchased from the Septentrio webshop:

- [AsteRx-m3 Pro](https://www.septentrio.com/en/products/gnss-receivers/oem-receiver-boards/asterx-m3-pro)
- [AsteRx-m3 Pro+](https://www.septentrio.com/en/products/gnss-receivers/oem-receiver-boards/asterx-m3-pro-plus)

Other PX4 supported devices from Septentrio:
- [mosaic-go evaluation kit](../../en/gps_compass/septentrio_mosaic-go.md)

## Interfaces

<div style="text-align: center;">

![Septentrio Robotics Interface Board Fritzing drawing](../../assets/hardware/gps/septentrio_sbf/RIB.png "Septentrio Robotics Interface Board Fritzing drawing")

</div>

### USB
_Connector type: micro USB type B._

The micro USB type B connector can be attached to a PC to power the receiver and to communicate with it over its USB
port.

### 44-pin header
_Connector type: SAMTEC TMM-122-03-S-D, 2-mm pitch._

The 44-pin header can be used to connect multiple GPIO devices. Please refer to
the [hardware manual](https://www.septentrio.com/system/files/support/asterx-m3_product_group_hardware_manual_2.2.0_1.pdf)
for the pinout.

> ### LED's

Reviewer: PX4 typically emits vehicle ready-to-fly information on GPS LEDs - is that setup supported? https://docs.px4.io/master/en/getting_started/led_meanings.html#ui-led

Author: Unfortunately, this is not supported, as an additional microcontroller is required to drive the LEDs.

I will submit a request to Septentrio to be able to support this though, as it is a useful feature.

The LED pins can be used to monitor the receiver status. They can be used to drive external LEDs (max drive current
10mA). It is assumed that the LED lights when the electrical level of the corresponding pin is high. The general-purpose
LED (GPLED pin) is configured with the setLEDMode command.

### Log Button Header

Putting a jumper on the LOG Button header (.100” vertical header) is equivalent to pressing a “log button”. The
interface board takes care of debouncing.

### PPS/Event Header

_Connector type: SAMTEC TMM-103-03-G-D, 2-mm pitch._

The 6-pin 2mm header next to the micro USB connector exposes the first PPS signal.

### Power Supply Options

When a USB cable is connected to the micro USB connector, the interface board is powered from the computer through the

> USB connector.Alternatively, the power can be applied from the PWR_IN pins of the 44-pin connector. The voltage range

Reviewer: ```suggestion
USB connector.
Alternatively, the power can be applied from the PWR_IN pins of the 44-pin connector. The voltage range

```

> When a USB cable is connected to the micro USB connector, the interface board is powered from the computer through the
> USB connector.Alternatively, the power can be applied from the PWR_IN pins of the 44-pin connector. The voltage range
> when powering from the PWR_IN pins is 4.5V to 30V. Power can be applied from both sources at the same time. On-board
> diodes prevent short circuits. The interface board provides the 3V3 supply to the AsteRx-m3 OEM receiver and a 5V DC
> voltage to the VANT pin of the AsteRx-m3 OEM.

Reviewer: ```suggestion
When a USB cable is connected to the micro USB connector, the interface board is powered from the computer through the USB connector.
Alternatively, the power can be applied from the PWR_IN pins of the 44-pin connector.
The voltage range when powering from the PWR_IN pins is 4.5V to 30V.
Power can be applied from both sources at the same time.
On-board diodes prevent short circuits. The interface board provides the 3V3 supply to the AsteRx-m3 OEM receiver and a 5V DC voltage to the VANT pin of the AsteRx-m3 OEM.
```

## Hardware setup

> <div style="text-align: center;">
> 
> ![Septentrio Robotics Interface Board wiring diagram](../../assets/hardware/gps/septentrio_sbf/RIB_wiring.png "Septentrio Robotics Interface Board wiring diagram")
> 
> </div>

Reviewer: ```suggestion
![Septentrio Robotics Interface Board wiring diagram](../../assets/hardware/gps/septentrio_sbf/RIB_wiring.png)
```

1. Make sure the receiver is powered with at least 3.3V. You can use the micro USB connector or the open ended supply (labeled "PWR & GND") on the 44 pin cable for this.
2. Connect one or two GNSS antennas to the external antenna ports on the AsteRx-i3 D board.
3. Connect the 44-pin cable to the AsteRx-i3 D board on RIB and connect the 6 pin JST GH connector to the UART & I2C B port on the Pixhawk 4 as shown in the diagram above.

:::note 
PX4 will ensure that the GNSS module is automatically configured however, if you have a dual antenna setup, it
is required to set the layout as accurately as possible in the web app.
:::

### Dual antenna

The attitude (heading/pitch) can be computed from the orientation of the baseline between the main and the aux1 GNSS
antennas.

> <div style="text-align: center;">
> 
> ![Multi-antenna attitude determination setup.]( ../../assets/hardware/gps/septentrio_sbf/multi-antenna_attitude_setup.png "Multi-antenna attitude determination setup.")
> 
> </div>

Reviewer: ```suggestion
![Multi-antenna attitude determination setup.]( ../../assets/hardware/gps/septentrio_sbf/multi-antenna_attitude_setup.png)

```

   These offsets can be compensated for with the **setAttitudeOffset** command.

:::note 
For optimal heading results, the two antennas should be seperated by at least 30cm / 11.8 in (ideally 50cm /
19.7in or more)

For additional configuration of the dual antenna setup, please refer to
our [Knowledge Base](https://customersupport.septentrio.com/s/article/Setting-up-and-configuring-the-AsteRx-i-compensating-for-a-non-default-antenna-orientation) or the [hardware manual](https://www.septentrio.com/system/files/support/asterx-i3_product_group_hardware_manual_1.0.0.pdf)
:::

### Web app

mosaic-H GPS/GNSS receiver module with heading comes with fully documented interfaces, commands and data messages. The
included GNSS receiver control and analysis software [RxTools](https://www.septentrio.com/en/products/software/rxtools)
allows receiver configuration, monitoring as well as data logging and analysis.

The receiver includes an intuitive web user interface for easy operation and monitoring allowing you to control the
receiver from any mobile device or computer. The web interface also uses easy-to-read quality indicators ideal to
monitor the receiver operation during the job at hand.

> <div style="text-align: center;">
> 
> ![Illustrative image for Septentrio mosaic-H GNSS Receiver Module Web User Interface (WebUI)]( ../../assets/hardware/gps/septentrio_sbf/Septentrio-mosaic-X5-H-T-CLAS-GNSS-Module-Receiver-WebUI.png "Septentrio web user interface")
> 
> </div>

Reviewer: ```suggestion
![Illustrative image for Septentrio mosaic-H GNSS Receiver Module Web User Interface (WebUI)]( ../../assets/hardware/gps/septentrio_sbf/Septentrio-mosaic-X5-H-T-CLAS-GNSS-Module-Receiver-WebUI.png "Septentrio web user interface")
```

## PX4 setup

> <div style="text-align: center;">
> 
> ![QGroundControl parameter settings](../../assets/hardware/gps/septentrio_sbf/QGC_param.png "QGroundControl parameter settings")
> 
> </div>

Reviewer: ```suggestion
![QGroundControl parameter settings](../../assets/hardware/gps/septentrio_sbf/QGC_param.png)
```

### Single antenna

> Edit the following parameters in the GPS tab:

Reviewer: `TELEM1` is generally used for the GCS telemetry stream, so doing this will stop the GCS connecting. Any reason you chose not to use GPS port for GPS? https://docs.px4.io/master/en/peripherals/mavlink_peripherals.html#telem1

Author: I have always used the `TELEM1` port because I only have a 6pin cable available. It is indeed better to use the `GPS` port, this should not be a problem. I will update the schematics and configuration.

> - [GPS_1_CONFIG](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_CONFIG): TELEM1
> - [GPS_1_GNSS](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_GNSS): 31
> - [GPS_1_PROTOCOL](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_PROTOCOL): Auto detect (
>   or SBF)
> - [SER_TEL1_BAUD](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#SER_TEL1_BAUD): 115200 8N1

Reviewer: These are local links:
```suggestion
- [GPS_1_CONFIG](../advanced_config/parameter_reference.md#GPS_1_CONFIG): TELEM1
- [GPS_1_GNSS](../advanced_config/parameter_reference.md##GPS_1_GNSS): 31
- [GPS_1_PROTOCOL](../advanced_config/parameter_reference.md#GPS_1_PROTOCOL): Auto detect (
  or SBF)
- [SER_TEL1_BAUD](../advanced_config/parameter_reference.md#SER_TEL1_BAUD): 115200 8N1

```

> Go to “Tools” &#8250; “Reboot Vehicle”

Reviewer: ```suggestion
Go to **Tools > Reboot Vehicle**
```

### Dual antenna

Edit the following parameters in the GPS tab:

> - [GPS_1_CONFIG](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_CONFIG): TELEM1
> - [GPS_1_GNSS](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_GNSS): 31
> - [GPS_1_PROTOCOL](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_1_PROTOCOL): Auto detect (
>   or SBF)
> - [SER_TEL1_BAUD](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#SER_TEL1_BAUD): 115200 8N1
> - [EKF2_AID_MASK](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#EKF2_AID_MASK): Use GPS & GPS
>   yaw fusion (129)
> - [GPS_YAW_OFFSET](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#GPS_YAW_OFFSET): set according
>   to your setup

Reviewer: ```suggestion
- [GPS_1_CONFIG](../advanced_config/parameter_reference.md#GPS_1_CONFIG): TELEM1
- [GPS_1_GNSS](../advanced_config/parameter_reference.md#GPS_1_GNSS): 31
- [GPS_1_PROTOCOL](../advanced_config/parameter_reference.md#GPS_1_PROTOCOL): Auto detect (or SBF)
- [SER_TEL1_BAUD](../advanced_config/parameter_reference.md#SER_TEL1_BAUD): 115200 8N1
- [EKF2_AID_MASK](https://docs.px4.io/master/en/advanced_config/parameter_reference.html#EKF2_AID_MASK): Use GPS & GPS yaw fusion (129)
- [GPS_YAW_OFFSET](../advanced_config/parameter_reference.md#GPS_YAW_OFFSET): set according to your setup
```

> Go to “Tools” &#8250; “Reboot Vehicle”

Reviewer: ```suggestion
Go to **Tools > Reboot Vehicle**
```
